libraries/chain/asset_evaluator.cpp

```diff
@@ -393,22 +393,61 @@ void_result asset_update_bitasset_evaluator::do_evaluate(const asset_update_bita
    return void_result();
 } FC_CAPTURE_AND_RETHROW( (o) ) }
 
-void_result asset_update_bitasset_evaluator::do_apply(const asset_update_bitasset_operation& o)
-{ try {
-   bool should_update_feeds = false;
-   // If the minimum number of feeds to calculate a median has changed, we need to recalculate the median
-   if( o.new_options.minimum_feeds != bitasset_to_update->options.minimum_feeds )
-      should_update_feeds = true;
+void_result asset_update_bitasset_evaluator::do_apply(const asset_update_bitasset_operation& op)
+{
+   try
+   {
+      auto& db_conn = db();
 
-   db().modify(*bitasset_to_update, [&](asset_bitasset_data_object& b) {
-      b.options = o.new_options;
+      // now do the actual modifications to the database object
+      db().modify(*bitasset_to_update, [op, &db_conn](asset_bitasset_data_object& bdo) {
 
-      if( should_update_feeds )
-         b.update_median_feeds(db().head_block_time());
-   });
+         // If the minimum number of feeds to calculate a median has changed, we need to recalculate the median
+         bool should_update_feeds = false;
+         if( op.new_options.minimum_feeds != bdo.options.minimum_feeds )
+            should_update_feeds = true;
 
-   return void_result();
-} FC_CAPTURE_AND_RETHROW( (o) ) }
+         // feeds must be reset if the backing asset is changed after hardfork 868
+         bool backing_asset_changed = false;
+         bool is_witness_or_committee_fed = false;
+         if (db_conn.get_dynamic_global_properties().next_maintenance_time > HARDFORK_CORE_868_TIME
+               && op.new_options.short_backing_asset != bdo.options.short_backing_asset)
+         {
+            backing_asset_changed = true;
+            should_update_feeds = true;
+            const asset_object& base_asset = op.asset_to_update(db_conn);
```

Comment:
For better performance, make this a member variable of `asset_update_bitasset_evaluator`, initialize it in `do_evaluate()`, so only need to query once.

Reply:
I could, but I would then have to pass a copy into the lambda, or add ``this`` to the capture list. Do we want to do that? Or am I misunderstanding your desired change?

Reply:
Check definition and use of `bitasset_to_update` variable. We can pass in a pointer when needed. 

By the way, as @pmconrad suggested, "the lambda has grown too large, its implementation should be moved into a separate static function".

Reply:
Done.